src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs

```diff
@@ -163,6 +167,57 @@ internal void WSHttpRequestReplyWithTransportMessageCertificateEchoString()
             }
         }
 
+#if NETCOREAPP3_1_OR_GREATER
+        [Fact, Description("transport-security-with-windows-authentication")]
+        [Trait("Category", "NetCoreOnly")] // Windows auth not supported on NetFx
+        internal void WSHttpRequestReplyWithTransportMessageImpersonateEchoString()
```

Comment:
I think you can get away with not conditionally compiling this test as the conditional piece is dealt with in the Startup class. The Trait on the test will prevent it running on NetFx.

Reply:
Will fix

Reply:
One thing to note, this test still fails locally for `net472`. The test request requires authentication since the SecurityMode is Transport, but the auth service is only added for netcore runtimes since we're using aspnetcore.

For now, I have removed the preprocessor condition from this test. Let me know if I should re-add it based on the above information.

Reply:
I spoke with the asp.net core team and this is expected. You could switch to using the HttpSys host instead of Kestrel and configuring it to use Windows auth and apparently that should work find in net472. HttpSys won't work on Linux, although you can reference it from Linux. It will just throw at runtime. Maybe you could make the code have different behavior based on what runtime framework it's running on and either use Kestrel with the existing code for .NET Core/.NET and use HttpSys configured for Windows auth on net472.

Reply:
If my understanding is correct, this test uses SecurityMode.Transport which requires an `https` endpoint. It seems like getting HttpSys to work with https [requires SSL cert registration](https://github.com/dotnet/aspnetcore/issues/4965#issuecomment-447352643).

Is there an effective way to handle this with unit tests? Or would it make more sense to throw a PlatformNotSupported exception if we're trying to use authentication with net472? Or let the test fail?

Reply:
If the machine has VS installed, it will generally have IIS express also installed, which sets up an ssl cert registration on some default ports. The IIS express test certificate is configured on ports 44300-44399.  So pick a port in that range somewhere and configure HttpSys to use that port for HTTPS.